samples/snippets/src/main/java/com/example/spanner/RestoreBackupWithEncryptionKey.java

```diff
@@ -0,0 +1,102 @@
+import com.google.cloud.spanner.BackupId;
+import com.google.cloud.spanner.Database;
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Restore;
+import com.google.cloud.spanner.RestoreInfo;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerExceptionFactory;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.encryption.EncryptionConfigs;
+import com.google.spanner.admin.database.v1.RestoreDatabaseMetadata;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class RestoreBackupWithEncryptionKey {
+
+  static void restoreBackupWithEncryptionKey() {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project";
+    String instanceId = "my-instance";
+    String databaseId = "my-database";
+    String backupId = "my-backup";
+    String kmsKeyName =
+        "projects/" + projectId + "/locations/<location>/keyRings/<keyRing>/cryptoKeys/<keyId>";
+
+    try (Spanner spanner =
+        SpannerOptions.newBuilder().setProjectId(projectId).build().getService()) {
+      DatabaseAdminClient adminClient = spanner.getDatabaseAdminClient();
+      restoreBackupWithEncryptionKey(
+          adminClient,
+          projectId,
+          instanceId,
+          backupId,
+          databaseId,
+          kmsKeyName);
+    }
+  }
+
+  static Void restoreBackupWithEncryptionKey(DatabaseAdminClient adminClient,
+      String projectId, String instanceId, String backupId, String restoreId, String kmsKeyName) {
+    final Restore restore = adminClient
+        .newRestoreBuilder(
+            BackupId.of(projectId, instanceId, backupId),
+            DatabaseId.of(projectId, instanceId, restoreId))
+        .setEncryptionConfig(EncryptionConfigs.customerManagedEncryption(kmsKeyName))
+        .build();
+    final OperationFuture<Database, RestoreDatabaseMetadata> operation = adminClient
+        .restoreDatabase(restore);
+
+    Database database;
+    try {
+      System.out.println("Waiting for operation to complete...");
+      database = operation.get(1600, TimeUnit.SECONDS);
+    } catch (ExecutionException e) {
+      // If the operation failed during execution, expose the cause.
+      throw SpannerExceptionFactory.asSpannerException(e.getCause());
+    } catch (InterruptedException e) {
+      // Throw when a thread is waiting, sleeping, or otherwise occupied,
+      // and the thread is interrupted, either before or during the activity.
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    } catch (TimeoutException e) {
+      // If the operation timed out propagates the timeout
+      throw SpannerExceptionFactory.propagateTimeout(e);
+    }
+
+    // Reload the metadata of the database from the server.
+    database = database.reload();
```

Comment:
nit: same as above, this should not be necessary, as the most recent metadata should have been returned by the operation.

Reply:
Thanks, fixed.